apps/schedules/lib/repo.ex

```diff
@@ -20,18 +20,23 @@ defmodule Schedules.Repo do
   @spec by_route_ids([Route.id_t()], Keyword.t()) :: [Schedule.t()] | {:error, any}
   def by_route_ids(route_ids, opts \\ []) when is_list(route_ids) do
     opts = Keyword.put_new(opts, :date, Util.service_date())
+    no_cache = Keyword.get(opts, :no_cache)
 
     @default_params
     |> Keyword.put(:route, Enum.join(route_ids, ","))
     |> Keyword.put(:date, Keyword.fetch!(opts, :date) |> to_string())
     |> add_optional_param(opts, :direction_id)
     |> add_optional_param(opts, :stop_sequences, :stop_sequence)
     |> add_optional_param(opts, :stop_ids, :stop)
-    |> cache(&all_from_params/1)
+    |> cache_condition(no_cache)
```

Comment:
I wonder if there's a way to get at this caching issue earlier, through the initial fetching of schedules. Like if we could get the  [If-Modified-Since](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/If-Modified-Since) header somehow before actually fetching the schedules and trips via the `V3Api.Schedules.all(params)` call, and use that to fetch from cache or not. But then why isn't the `cache` function already doing this? Hm 🤔 🤔

Reply:
Right??  I'm a little baffled by that bit.  Maybe we could do a partner programming sesh to look at how caching works?

Reply:
Yeah. Hmm. Can you start a spinoff investigation ticket for this? I'd like to merge in this PR in the meantime (who knows, maybe it's already the whole fix we need!)

Reply:
Great!  Ticket made:  https://app.asana.com/0/385363666817452/1200531793045978